src/AutoRest.CSharp/Input/SchemaUsageProvider.cs

```diff
@@ -18,7 +18,18 @@ public SchemaUsageProvider(CodeModel codeModel)
 
                 if (usage != null)
                 {
-                    Apply(objectSchema, (SchemaTypeUsage) Enum.Parse(typeof(SchemaTypeUsage), usage, true));
+                    var schemaTypeUsage = (SchemaTypeUsage)Enum.Parse(typeof(SchemaTypeUsage), usage, true);
+
+                    if (schemaTypeUsage.HasFlag(SchemaTypeUsage.Converter))
+                    {
+                        Apply(objectSchema, SchemaTypeUsage.Converter, false);
+                        schemaTypeUsage &= ~SchemaTypeUsage.Converter;
+                    }
+
+                    if (!schemaTypeUsage.Equals(default(SchemaTypeUsage)))
```

Comment:
WHy do we need this check?

Reply:
I added this check if user only passes "x-csharp-usage": "converter", then this check will prevent us calling `Apply` method again. Although there will be no effect even if we remove this check because `Apply` method will return from [here](https://github.com/Azure/autorest.csharp/blob/194731d242ba2cfa55886aa8ed70493f41583d32/src/AutoRest.CSharp/Input/SchemaUsageProvider.cs#L105).

Reply:
Yeah, let's simplify logic here, the `Apply` should already handle this.